ncm-ncd 16.6.0-rc3: cannot acquire ncm-ncd lock during upgrade

During Quattor client upgrade through `ncm-cdispd`, `ncd-ncd` is run 3 times: this is due to `ncm-cdispd` service restart during upgrade of `ncm-cdispd` and `cdp-listend`. Second and third `ncm-ncd` instances are blocked by `ncm-ncd` lock, until the previous instance has completed.

It seems that the new `CAF::Lock` in 16.6.0-rc3 causes the third instance to exit before running with the message:
```
2016/07/11-20:16:41 [ERROR] cannot acquire lock after 10 tries: /var/lock/quattor/ncm-ncd
```

I don't think this was the case before (it was just waiting until the process whose pid was in the lock file had exited). It should not be a problem (in fact only one restart of `ncm-cdispd` is normally enough) but if for some reasons `ncm-ncd` takes longer to complete, there is may the risk that it will not be restarted (while the running one will be terminated at the end of `ncm-ncd` run), leaving the system unmanaged.

Is it possible to confirm that this is not a problem or to ensure a longer (infinite?) wait?

Probably a side effect of #64. May be good to fix it or revert #64 before releasing 16.6.0-rc3, even if not strictly speaking a showstopper...

Would be good to look at this issue before releasing [16.6.0-rc3](https://github.com/quattor/release/issues/250), even if not strictly speaking a showstopper...

should be easy to test: take the lock on command line , and restart ncm-cdispd

```bash
PERL5LIB=/usr/lib/perl perl -e 'use CAF::Lock; my $l = CAF::Lock->new("/var/lock/quattor/ncm-ncd"); $l->set_lock(); sleep 1000'
```

`ncm-cdispd` stays running

extract from ncm-cdispd log

```
2016/08/18-21:18:37 [VERB] [VERB] accessing CCM cache manager..
2016/08/18-21:18:37 [VERB] [VERB] getting locked CCM configuration..
2016/08/18-21:18:37 [VERB] [VERB] checking for ncm-ncd locks...
2016/08/18-21:18:37 [VERB] [DEBUG] Empty lockfile /var/lock/quattor/ncm-ncd found: no old-style lock
2016/08/18-21:18:37 [VERB] [DEBUG] flock on /var/lock/quattor/ncm-ncd gave has_lock 0
...
2016/08/18-21:23:07 [VERB] [VERB] lock file is already held, try 10 out of 10 (timeout 30)
2016/08/18-21:23:37 [VERB] [DEBUG] Empty lockfile /var/lock/quattor/ncm-ncd found: no old-style lock
2016/08/18-21:23:37 [VERB] [DEBUG] flock on /var/lock/quattor/ncm-ncd gave has_lock 0
2016/08/18-21:23:37 [VERB] [ERROR] cannot acquire lock after 11 tries (timeout 30): /var/lock/quattor/ncm-ncd
...
2016/08/18-21:23:37 [WARN] ncm-ncd finished with status: 255 (ec 65280, some configuration modules failed to run successfully)
2016/08/18-21:23:37 [DEBUG] process_signal(): no delayed signal to process
2016/08/18-21:23:37 [DEBUG] Initializing $last_ncd_status to init_components status (1)
2016/08/18-21:23:37 [DEBUG] CID of reference configuration set to 49
2016/08/18-21:23:37 [DEBUG] checking for new profiles ...
2016/08/18-21:23:37 [DEBUG] CID of last profile processed: 49
2016/08/18-21:23:37 [VERB] no new profile found, sleeping
2016/08/18-21:23:37 [DEBUG] sleep for 60 seconds
...
```

--- a/src/main/scripts/ncm-ncd
+++ b/src/main/scripts/ncm-ncd
@@ -89,7 +89,7 @@ sub app_options()
         {
             NAME    => 'retries=i',
             HELP    => 'number of retries if ncd is locked',
-            DEFAULT => 10
+            DEFAULT => 30
         },
 
         {

